base/linalg/rowvector.jl

```diff
@@ -73,18 +76,22 @@ julia> transpose(v)
 ```
 """
 @inline transpose(vec::AbstractVector) = RowVector(vec)
-@inline ctranspose{T}(vec::AbstractVector{T}) = RowVector(conj(vec))
+@inline ctranspose{T}(vec::AbstractVector{T}) = RowVector(_conj(vec))
 @inline ctranspose{T<:Real}(vec::AbstractVector{T}) = RowVector(vec)
 
 @inline transpose(rowvec::RowVector) = rowvec.vec
+@inline transpose(rowvec::ConjRowVector) = copy(rowvec.vec) # remove the ConjArray wrapper from any raw vector
```

Comment:
The logic was to make sure that vectors are always unwrapped, so that calls to BLAS are generally used. This is partly because the PR doesn't create a bunch of new methods for `*(conjarray, array)`, etc. We also have "view semantic" for row vectors and "copy semantic" for vectors, which seemed more consistent than views for vectors and rowvectors and copies for everything else.

(That is, from the user's perspective, only the new `RowVector` type gets the view semantic, conjugated or not. The old types behave as before, so *hopefully* this is more backward compatible. Ideally these changes would have been made at the same time as `TransposedMatrix` to make all of this more consistent...)